### PR TITLE
optimize catalog fetch

### PR DESCRIPTION
**Pull request by amy**
Addresses this issue: https://github.com/rancher/rancher/issues/9057

### CODE REVIEW

**Comment by amy** [comment]
cc/ @cjellick @joshwget

**Comment by vincent99** [comment]
@amy @cjellick I don't know if this is better than the current behavior or not, but this does not address #9057 as was described.  `git remote-ls http://github.com/rancher/rancher-catalog master` is about 67kb of network traffic per check (= about half a gig/install/month).

Using the API (which I described and Denise copied into the issue) to see if the SHA is current is ~90% less at 5kb per check.

**Comment by amy** [comment]
@vincent99  Ah I see. This is my bad. Didn't realize that `git ls-remote` took that much traffic. Just went ahead and did it because I saw it being used in that git repo already instead of what was noted in the git issue.

**Comment by amy** [comment]
Suggestions of how to add integration tests for this are welcome.

File: git/git.go [hunk 1]
```diff
@@ -26,6 +27,51 @@ func HeadCommit(path string) (string, error) {
...
+	if strings.HasPrefix(url, "https://github.com") && len(splitURL) > 4 {
+		org := splitURL[3]
+		repo := strings.TrimSuffix(splitURL[4], ".git")
+
+		return fmt.Sprintf("https://api.github.com/repos/%s/%s/commits/%s", org, repo, branch)
+	} else if strings.HasPrefix(url, "https://git.rancher.io") && len(splitURL) > 3 {
+		org := "rancher"
+		repo := strings.TrimSuffix(splitURL[3], ".git")
+
+		return fmt.Sprintf("https://api.github.com/repos/%s/%s/commits/%s", org, repo, branch)
```
Comment (vincent99) [inline git/git.go]: This needs to go to an endpoint we create on the git mirror that works the same as GitHub's.  The main point of git.rancher.io is to be an endpoint people can whitelist (or not have blacklisted in the first place) that *isn't* the entirety of the GitHub world.

**Comment by vincent99** [comment]
LGTM once tested against mirror

**Comment by deniseschannon** [comment]
per offline convo with @amy, she's tested against the mirror.

**Pull request merged**